solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,495 @@
+########################################################################################################################
+# Input variables
+########################################################################################################################
+
+##############################################################
+# General Configuration
+##############################################################
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud api key."
+  sensitive   = true
+}
+
+variable "prefix" {
+  type        = string
+  description = "The prefix to add to all resources that this solution creates (e.g `prod`, `test`, `dev`). To not use any prefix value, you can set this value to `null` or an empty string."
+  nullable    = true
+  validation {
+    condition = (var.prefix == null ? true :
+      alltrue([
+        can(regex("^[a-z]{0,1}[-a-z0-9]{0,14}[a-z0-9]{0,1}$", var.prefix)),
+        length(regexall("^.*--.*", var.prefix)) == 0
+      ])
+    )
+    error_message = "Prefix must begin with a lowercase letter, contain only lowercase letters, numbers, and - characters. Prefixes must end with a lowercase letter or number and be 16 or fewer characters."
+  }
+}
+
+variable "existing_resource_group_name" {
+  type        = string
+  description = "The name of an existing resource group to provision the cluster."
+  default     = "Default"
+}
+
+variable "cluster_resource_tags" {
+  type        = list(string)
+  description = "Metadata labels describing this cluster deployment, i.e. test."
+  default     = []
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the resources created by the module."
+  default     = []
+}
+
+##############################################################
+# Cluster Related
+##############################################################
+
+variable "cluster_name" {
+  type        = string
+  description = "The name of the new IBM Cloud OpenShift Cluster. If a `prefix` input variable is specified, it is added to this name in the `<prefix>-value` format."
+  default     = "openshift"
+}
+
+variable "ocp_version" {
+  type        = string
+  description = "Version of the OCP cluster to provision."
+  default     = "4.17"
+}
+
+variable "ocp_entitlement" {
+  type        = string
+  description = "Value that is applied to the entitlements for OCP cluster provisioning."
+  default     = null
+}
+
+variable "cluster_ready_when" {
+  type        = string
+  description = "The cluster is ready when one of the following: MasterNodeReady (not recommended), OneWorkerNodeReady, Normal, IngressReady."
+  default     = "IngressReady"
+}
+
+variable "enable_ocp_console" {
+  description = "Flag to specify whether to enable or disable the OpenShift console."
+  type        = bool
+  default     = true
```

Comment:
This should default to null, just like the module nodes. And update the var description to match the module

Reply:
Done